src/theory/quantifiers/proof_checker.cpp

```diff
@@ -120,6 +121,41 @@ Node QuantifiersProofRuleChecker::checkInternal(
         body.substitute(vars.begin(), vars.end(), subs.begin(), subs.end());
     return inst;
   }
+  else if (id == PfRule::ALPHA_EQUIV)
+  {
+    Assert(children.empty());
+    if (args[0].getKind() != kind::FORALL)
+    {
+      return Node::null();
+    }
+    // arguments must be equalities that are bound variables that are
+    // pairwise unique
+    std::unordered_set<Node> allVars[2];
+    std::vector<Node> vars;
+    std::vector<Node> newVars;
+    for (size_t i = 1, nargs = args.size(); i < nargs; i++)
+    {
+      if (args[i].getKind() != kind::EQUAL)
+      {
+        return Node::null();
+      }
+      for (size_t j = 0; j < 2; j++)
+      {
+        Node v = args[i][j];
+        if (v.getKind() != kind::BOUND_VARIABLE
+            || allVars[j].find(v) != allVars[j].end())
```

Comment:
Feels a bit odd to use sets when you are enforcing uniqueness. Maybe just use vector?

Reply:
I think set is cleaner in this case.

Reply:
ok